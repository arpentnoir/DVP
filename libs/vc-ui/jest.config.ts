/* eslint-disable */
export default {
  displayName: 'vc-ui',
  preset: '../../jest.preset.js',
  setupFiles: ['./jest.setup.ts'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/vc-ui',
  moduleNameMapper: {
    '^react-pdf': 'react-pdf/dist/umd/entry.jest',
  },
};
