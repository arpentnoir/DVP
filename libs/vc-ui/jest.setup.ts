import 'jest-canvas-mock';
import '@testing-library/jest-dom';

const mockClipboard = {
  writeText: jest.fn(),
};

(global.navigator.clipboard as any) = mockClipboard;
