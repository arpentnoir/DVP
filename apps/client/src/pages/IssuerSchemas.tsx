import { Card, Text } from '@dvp/vc-ui';
import { Box, Stack, Switch } from '@mui/material';
import { useParams } from 'react-router-dom';
import { BaseLayout } from '../layouts';

// TODO: Replace with real data
const schemas = [
  {
    name: 'Schema 1',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
  {
    name: 'Schema 2',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
  {
    name: 'Schema 3',
    text: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.',
  },
];

export const IssuerSchemas = () => {
  const { issuerName } = useParams();

  return (
    <BaseLayout title="Issuer Schemas">
      <Box paddingBottom="40px">
        <Text variant="h4" fontWeight="bold">
          {issuerName}
        </Text>
      </Box>

      <Box>
        <Text variant="body1" fontWeight="bold">
          Document schemas
        </Text>
        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</Text>
        <Stack
          direction={{ xs: 'column', sm: 'row' }}
          spacing="8px"
          paddingTop="40px"
        >
          {schemas.map((schema) => (
            <Card
              key={schema.name}
              name={schema.name}
              text={schema.text}
              headerAction={<Switch />}
            />
          ))}
        </Stack>
      </Box>
    </BaseLayout>
  );
};
