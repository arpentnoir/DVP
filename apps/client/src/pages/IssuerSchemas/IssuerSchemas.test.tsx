import { render } from '@dvp/vc-ui';
import { IssuerSchemas } from './IssuerSchemas';

describe('IssuerSchemas', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<IssuerSchemas />);

    expect(baseElement).toMatchSnapshot();
  });
});
