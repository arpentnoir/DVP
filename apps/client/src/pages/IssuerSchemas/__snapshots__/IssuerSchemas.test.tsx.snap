// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuerSchemas should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-disableGutters css-1y9y4eh-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-m31y3a"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1e0njhx-MuiTypography-root"
          tabindex="0"
        />
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-1ey9xtw-MuiTypography-root"
          tabindex="0"
        >
          Document schemas
        </h3>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          tabindex="0"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        </p>
        <div
          class="css-u5ktxm-MuiStack-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-nblda9-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="css-3y1137-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
                  tabindex="0"
                >
                  Schema 1
                </p>
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body2 css-ik0l5y-MuiTypography-root"
                tabindex="0"
              >
                It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
              </p>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-nblda9-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="css-3y1137-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
                  tabindex="0"
                >
                  Schema 2
                </p>
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body2 css-ik0l5y-MuiTypography-root"
                tabindex="0"
              >
                It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
              </p>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-nblda9-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="css-3y1137-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
                  tabindex="0"
                >
                  Schema 3
                </p>
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body2 css-ik0l5y-MuiTypography-root"
                tabindex="0"
              >
                It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
